rudder-core/src/main/scala/com/normation/rudder/services/ClearCacheService.scala

```diff
@@ -0,0 +1,108 @@
+import com.normation.rudder.repository.CachedRepository
+import com.normation.rudder.batch.{AsyncDeploymentAgent, AutomaticStartDeployment}
+import com.normation.rudder.domain.eventlog.ClearCacheEventLog
+import net.liftweb.common.{EmptyBox, Full, Loggable}
+import net.liftweb.http.S
+import com.normation.rudder.repository.EventLogRepository
+import com.normation.rudder.services.policies.nodeconfig.NodeConfigurationHashRepository
+import com.normation.utils.StringUuidGenerator
+import net.liftweb.common.Box
+
+trait ClearCacheService {
+
+  /*
+  * This one only clear the "node configuration" cache. That will
+  * force a full regeneration of all policies
+  */
+ def clearNodeConfigurationCache(storeEvent: Boolean, actor: EventActor) : Box[Unit]
+
+  /*
+  * This method clear all caches, which are:
+  * - node configurations (force full regen)
+  * - cachedAgentRunRepository
+  * - recentChangesService
+  * - reportingServiceImpl
+  * - nodeInfoServiceImpl
+  */
+ def action(actor: EventActor) : Box[String]
+}
+
+class ClearCacheServiceImpl(
+      nodeConfigurationService : NodeConfigurationHashRepository
+    , asyncDeploymentAgent     : AsyncDeploymentAgent
+    , eventLogRepository       : EventLogRepository
+    , uuidGen                  : StringUuidGenerator
+    , clearableCache           : Seq[CachedRepository]
+) extends ClearCacheService with Loggable {
+
+  def clearNodeConfigurationCache(storeEvent: Boolean = true, actor: EventActor) = {
+    nodeConfigurationService.deleteAllNodeConfigurations match {
+      case eb: EmptyBox =>
+        (eb ?~! "Error while clearing node configuration cache")
+      case Full(set) =>
+        if (storeEvent) {
+          val modId = ModificationId(uuidGen.newUuid)
+          eventLogRepository.saveEventLog(
+            modId
+            , ClearCacheEventLog(
+              EventLogDetails(
+                modificationId = Some(modId)
+                , principal = actor
+                , details = EventLog.emptyDetails
+                , reason = Some("Node configuration cache deleted on user request")
+              )
+            )
+          ) match {
+            case eb: EmptyBox =>
+              val e = eb ?~! "Error when logging the cache event"
+              logger.error(e.messageChain)
+              logger.debug(e.exceptionChain)
+            case _ => //ok
+          }
+          logger.debug("Deleting node configurations on user clear cache request")
+          asyncDeploymentAgent ! AutomaticStartDeployment(modId, actor)
+        }
+        Full(set)
+    }
+  }
+
+  def action(actor: EventActor) = {
+
+    S.clearCurrentNotices
+    val modId = ModificationId(uuidGen.newUuid)
+
+    //clear agentRun cache
+    clearableCache.foreach {_.clearCache}
+
+    //clear node configuration cache
+    (for {
+      set <- clearNodeConfigurationCache(storeEvent = false, actor)
+      _ <-  eventLogRepository.saveEventLog(
```

Comment:
Can you indent it correctly please ? exactly like it was done before https://github.com/Normation/rudder/pull/1946/files#diff-a778e7d11c9c87410a1c06ef983dbc53L120